File: 2026/04-April/10-Friday/solution.java
import java.util.*;

class Solution {
    public int minimumDistance(int[] nums) {
        Map<Integer, List<Integer>> map = new HashMap<>();
 
        for (int i = 0; i < nums.length; i++) {
            map.computeIfAbsent(nums[i], k -> new ArrayList<>()).add(i);
        }

        int ans = Integer.MAX_VALUE;

        for (List<Integer> list : map.values()) {
            for (int i = 0; i <= list.size() - 3; i++) {
                ans = Math.min(ans, 2 * (list.get(i + 2) - list.get(i)));
            }
        }

        return ans == Integer.MAX_VALUE ? -1 : ans;
    }
}

File: 2026/04-April/10-Friday/solution.js
var minimumDistance = function(nums) {
    let map = new Map();

    for (let i = 0; i < nums.length; i++) {
        if (!map.has(nums[i])) map.set(nums[i], []);
        map.get(nums[i]).push(i);
    }

    let ans = Infinity;

    for (let arr of map.values()) {
        for (let i = 0; i + 2 < arr.length; i++) {
            ans = Math.min(ans, 2 * (arr[i+2] - arr[i]));
        }
    }

    return ans === Infinity ? -1 : ans;
};
